### PR TITLE
Imports Polling

### PR DESCRIPTION
**Pull request by KirtiGautam**
- Added polling in imports.
- Added a loading buffer bar.

### CODE REVIEW

File: src/app/qbo/sync-export/sync/sync.component.ts [hunk 1]
```diff
@@ -29,18 +30,29 @@ export class SyncComponent implements OnInit {
...
       that.isExpensesSyncing = false;
       that.snackBar.open('Import Failed');
       that.errorOccurred = true;
     });
   }
 
+  checkSyncStatus() {
+    const that = this;
+    interval(3000).pipe(
+      switchMap(() => from(that.taskService.getAllTasks('ALL'))),
```
Comment (ashwin1111) [inline src/app/qbo/sync-export/sync/sync.component.ts:45]: Nilesh had some changes in task service in this area, please check that. Pass required type and status. Refer to latest master export page.

File: src/app/qbo/sync-export/sync/sync.component.ts [hunk 2]
```diff
@@ -29,18 +30,30 @@ export class SyncComponent implements OnInit {
...
       that.snackBar.open('Import Failed');
       that.errorOccurred = true;
     });
   }
 
+  checkSyncStatus() {
+    const that = this;
+    const taskType = ['FETCHING_EXPENSES'];
+    interval(3000).pipe(
+      switchMap(() => from(that.taskService.getAllTasks('ALL', [], taskType))),
```
Comment (ashwin1111) [inline src/app/qbo/sync-export/sync/sync.component.ts:46]: only `IN_PROGRESS` should be enough instead if `ALL` right? please verify this.
Don't have to send []
Can send something like this `from(status='IN_PROGRESS', taskType=taskType)`

File: src/app/qbo/sync-export/sync/sync.component.ts [hunk 3]
```diff
@@ -29,18 +30,30 @@ export class SyncComponent implements OnInit {
...
+      takeWhile((response) => response.results.filter(task => task.status === 'IN_PROGRESS'  && task.type === 'FETCHING_EXPENSES').length > 0, true)
```
Comment (ashwin1111) [inline src/app/qbo/sync-export/sync/sync.component.ts:47]: if you pass `IN_PROGRESS` and `FETCHING_EXPENSES` from front end, this check might not be required, can just check the count only

File: src/app/qbo/sync-export/sync/sync.component.ts [hunk 4]
```diff
@@ -29,18 +30,30 @@ export class SyncComponent implements OnInit {
...
+    ).subscribe((res) => {
+      if (res.results.filter(task => task.status === 'COMPLETE'  && task.type === 'FETCHING_EXPENSES').length === 1) {
```
Comment (ashwin1111) [inline src/app/qbo/sync-export/sync/sync.component.ts:49]: check this part too, if you modify above ones

File: src/app/qbo/sync-export/sync/sync.component.ts (hunk 2 above)
Comment (KirtiGautam) [reply src/app/qbo/sync-export/sync/sync.component.ts:46]: Unfortunately, Kwargs are not possible in js. There is something called named args but that requires changing the params in method itself. [Link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments)

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**